src/charm.py

```diff
@@ -931,9 +931,19 @@ def _update_kratos_info_relation_data(self, event: RelationEvent) -> None:
         configmaps_namespace = self.model.name
         mfa_enabled = self.config.get("enforce_mfa")
 
+        public_url = self._public_url
+        if public_url is None:
+            self.unit.status = BlockedStatus(
+                "Cannot add external provider without an external hostname. Please "
```

Comment:
It's just a nitpick, but the status message is a bit misleading. Adding an "internal" idp would also fail without the external url. Perhaps we could change it to sth like:
```suggestion
                "Cannot send integration data without an external hostname. Please "
```